sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/faultinjection/ExcludeRegionWithFaultInjectionTests.java

```diff
@@ -0,0 +1,1886 @@
+                            this.chooseFirstRegion.apply(this.preferredRegions)
+                    ))
+                },
+                {
+                    "503/21008_firstRegion_beforeMutation_excludeFirstRegion_afterMutation_excludeNoRegions",
+                    new MutationTestConfig()
+                        .withChooseInitialExclusionRegions(this.chooseFirstRegion)
+                        .withChooseFaultInjectionRegions(this.chooseFirstRegion)
+                        .withFaultInjectionOperationType(FaultInjectionOperationType.READ_ITEM)
+                        .withFaultInjectionServerErrorType(FaultInjectionServerErrorType.SERVICE_UNAVAILABLE)
+                        .withDataPlaneOperationExecutor(readItemCallback)
+                        .withRegionExclusionMutator((regions) -> new ArrayList<>())
+                        .withExpectedResultBeforeMutation(new ExpectedResult(
+                            HttpConstants.StatusCodes.OK,
+                            HttpConstants.SubStatusCodes.UNKNOWN,
+                            this.chooseLastRegion.apply(this.preferredRegions)
+                        ))
+                        .withExpectedResultAfterMutation(new ExpectedResult(
+                            HttpConstants.StatusCodes.OK,
+                            HttpConstants.SubStatusCodes.UNKNOWN,
+                            this.chooseAllRegions.apply(this.preferredRegions)
+                    ))
+                },
+                {
+                    "500/0_allRegions_beforeMutation_excludeFirstRegion_afterMutation_excludeSecondRegion",
+                    new MutationTestConfig()
+                        .withChooseInitialExclusionRegions(this.chooseFirstRegion)
+                        .withChooseFaultInjectionRegions(this.chooseAllRegions)
+                        .withFaultInjectionOperationType(FaultInjectionOperationType.READ_ITEM)
+                        .withFaultInjectionServerErrorType(FaultInjectionServerErrorType.INTERNAL_SERVER_ERROR)
+                        .withDataPlaneOperationExecutor(readItemCallback)
+                        .withRegionExclusionMutator(this.chooseLastRegion)
+                        .withExpectedResultBeforeMutation(new ExpectedResult(
+                            HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR,
+                            HttpConstants.SubStatusCodes.UNKNOWN,
+                            this.chooseSecondRegion.apply(this.preferredRegions)
+                        ))
+                        .withExpectedResultAfterMutation(new ExpectedResult(
+                            HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR,
+                            HttpConstants.SubStatusCodes.UNKNOWN,
+                            this.chooseFirstRegion.apply(this.preferredRegions)
+                    ))
+                }
+            };
+        } else if (this.preferredRegions.size() == 3) {
```

Comment:
I though for multi-master group, there will only be 2 regions, will this part ever be executed?

Reply:
Correct - multi-master group is configured with 2 regions. This code path is tested against an account with 3 write regions.

Reply:
Why don't we add one more region to the CI coverage? So that this condition can be always tested in the CI without having to run it manually separately?